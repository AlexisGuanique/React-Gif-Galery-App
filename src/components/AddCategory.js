import React, { useState } from 'react'
import PropTypes from 'prop-types';


// Creamos componente para aÃ±adir una categoria
export const AddCategory = ( { setCategories } ) => {
    
    // Definimos el estado inicial de nuestro value
    const [InputValue, setInputValue] = useState('');


    // Extraemos el valor a guardar
    const handlInputChange = (e) => {
        // console.log(e.target.value);
        setInputValue(e.target.value);

        // console.log('HandleInputChange llamado')
    }


    // De esta manera evitamos que refresque el formulario de la manera que viene por defecto
    const handleSubmit = (e) => {
        e.preventDefault()

        // console.log('handleSubmit')

        // Validamos que el valor ingresado sea una palabra mayor de 2 letras
        if ( InputValue.trim().length > 1) {

            setCategories( cats => [ InputValue, ...cats ]);
            setInputValue('');
        }
    };
  

    // Lo que renderiza nuestro componente
    return (
        <form onSubmit={ handleSubmit }>
            <p>{ InputValue }</p>
            <input 
                type="text"
                value={ InputValue }
                onChange= { handlInputChange }         
            />
        </form>
  )
};

// De esta manera decimos que la Prop setCategories es requerida para utilizar nuestro componente AddCategory
AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};
