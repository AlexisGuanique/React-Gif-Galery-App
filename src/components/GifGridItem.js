import React from 'react';
import { PropTypes } from 'prop-types';

export const GifGridItem = ( {title, url } ) => {

    // console.log( id, title, url );

    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={ url } alt={ title }/>
            <p className='title'>{ title }</p>
        </div>
    )
};


// Configuracion de los PropTypes, de esta manera indicamos los lineamientos para trabajar con nuestro componente
GifGridItem.protoType = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
};
