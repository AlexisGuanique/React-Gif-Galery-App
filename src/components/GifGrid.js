import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';


// Componente que consume la api_key y muestra los gif
export const GifGrid = ( { category } ) => {


    // Utilizamos nuestro custom hook
    const { data:images, loading } = useFetchGifs( category );

    console.log(images);

    return (
        <>
            <h2 className='animate__animated animate__fadeIn'> { category } </h2>

            { loading && <p className='animate__animated animate__flash'>Cargando...</p> }
            
            <div className='card-grid'>
                {
                    images.map( img => (
                        <GifGridItem
                            key={ img.id } 
                            { ...img }
                        />
                    ))
                }
            </div>
        </>
    )
};

