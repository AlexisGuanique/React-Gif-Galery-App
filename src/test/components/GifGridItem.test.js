import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from "../../components/GifGridItem"


describe('Pruebas en <GifGridItem />', () => {

    // Variables globales dentro de la prueba
    const title = 'Titulo';
    const url = 'https://holamundo.jpg';
    const wrapper = shallow( <GifGridItem title={ title } url={ url }/> ) 



    test('Debe de mostrar el componente correctamente', () => {

        expect( wrapper ).toMatchSnapshot();

    });

    test('Debe tener un parrafo con el titulo', () => {

        const p = wrapper.find( 'p' );

        expect( p.text().trim() ).toBe( title );
    });

    test('Debe de tener la imagen igual al url y alt de los props', () => {

        const img = wrapper.find( 'img' );

        expect( img.prop( 'src' )).toBe( url );
        expect( img.prop( 'alt' )).toBe( title );
    });

    test('Debe de tener la clase animate__fadeIn', () => {

        const div = wrapper.find( 'div' );
        // Manera larga de hacerlo
        // expect( div.prop('className')).toBe( 'card animate__animated animate__fadeIn' );

        // Manera corta y mas eficiente

        // La palabra reservada includes me ayuda a consultar si ese string incluye una palabra u oracion en especifico
        const className = div.prop('className');
        expect( className.includes( 'animate__fadeIn') ).toBe( true );

    })


 

})


