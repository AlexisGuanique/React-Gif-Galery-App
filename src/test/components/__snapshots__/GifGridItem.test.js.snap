// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://holamundo.jpg"
  />
  <p
    className="title"
  >
    Titulo
  </p>
</div>
`;
